StockServiceFactory/src/main/java/edu/mbrooks/advancedjava/StockFactory/StockFactory.java

```diff
@@ -49,24 +57,178 @@ public void addQuote(StockQuote q) {
 
         List<StockQuote> returnList = new ArrayList<>();
 
         for (StockQuote q : this.historyQuotes){
-            if (q.getDateOfQuote().after(from) && q.getDateOfQuote().before(until)) {
-                returnList.add(q);
-            } else if (q.getDateOfQuote().compareTo(from) == 0) {
-                returnList.add(q);
-            } else if (q.getDateOfQuote().compareTo(until) == 0) {
-                returnList.add(q);
+
+            if (isQuoteInDateRange(q.getDateOfQuote(), from, until)) {
+
+                switch (frequency) {
+                    case HOURLY:
+                        // interval needs to be some key... Using YEAR+Month+Day+Hour to get something unique
+                        if  (isQuoteInDateRange(q.getDateOfQuote(),q.getDateOfQuote(),q.getDateOfQuote())) {
+                            fuzzyTime =  q.getDateOfQuote().get(Calendar.YEAR) + q.getDateOfQuote().get(Calendar.DAY_OF_YEAR) + q.getDateOfQuote().get(Calendar.HOUR_OF_DAY);
+                            if (fuzzyTime != intervalCheck) {
+                                intervalCheck = fuzzyTime;
+                                returnList.add(q);
+                            }
+                        }
+                        break;
+                     case DAILY:
+                            if (q.getDateOfQuote().get(Calendar.DAY_OF_YEAR)  != intervalCheck) {
+                                intervalCheck = q.getDateOfQuote().get(Calendar.DAY_OF_YEAR);
+                                returnList.add(q);
+                            }
+                            break;
+                    case WEEKLY:
+                       if (q.getDateOfQuote().get(Calendar.WEEK_OF_YEAR) != intervalCheck) {
+                            intervalCheck = q.getDateOfQuote().get(Calendar.WEEK_OF_YEAR);
+                            returnList.add(q);
+                        }
+                        break;
+                    case MONTHLY:
+                        if (q.getDateOfQuote().get(Calendar.MONTH) != intervalCheck) {
+                            intervalCheck = q.getDateOfQuote().get(Calendar.MONTH);
+                            returnList.add(q);
+                        }
+                        break;
+                    case QUARTERLY:
+
+                        month =  q.getDateOfQuote().get(Calendar.MONTH);
+
+                        if (q.getDateOfQuote().get(Calendar.YEAR) != intervalCheck) {
+                            // flipping this on the annual rollover
+                            intervalCheck = q.getDateOfQuote().get(Calendar.YEAR);
+                            // reset quarterly flags on the turn of the year
+                            q1Check = false;
+                            q2Check = false;
+                            q3Check = false;
+                            q4Check = false;
+                        }
+
+                        switch (whatQuarter(q.getDateOfQuote())) {
+                            case Q1:
```

Comment:
again, not sure this is needed at all.

Reply:
Again, looking to make sure I'm getting a single quote for a given quarter, and only return a single quote per quarter.